core/samples/event_system.cpp

```diff
@@ -0,0 +1,85 @@
+#include <cubos/core/log.hpp>
+#include <cubos/core/ecs/event_pipe.hpp>
+#include <cubos/core/ecs/event_writer.hpp>
+#include <cubos/core/ecs/event_reader.hpp>
+
+using namespace cubos::core::ecs;
+using namespace cubos::core::memory;
+
+int main()
+{
+    cubos::core::initializeLogger();
+
+    struct MyEvent
+    {
+        enum Mask
+        {
+            KEY_EVENT = 1 << 0,   // 0001
+            MOUSE_EVENT = 1 << 1, // 0010
+            WHEEL_EVENT = 1 << 2, // 0100
+            ALL = ~0u,            // 1111
+        };
+
+        char a;
+        int data; // random data member
+    };
+
+    auto pipe = EventPipe<MyEvent>();
+    auto writer = EventWriter<MyEvent>(pipe);
+
+    writer.push(MyEvent{.data = 4}, MyEvent::Mask::KEY_EVENT);
+    writer.push(MyEvent{.data = 1}, MyEvent::Mask::WHEEL_EVENT);
+    writer.push(MyEvent{.data = 2}, MyEvent::Mask::MOUSE_EVENT);
+    writer.push(MyEvent{.data = 6}, MyEvent::Mask::MOUSE_EVENT);
+    writer.push(MyEvent{.data = 15});
+    writer.push(MyEvent{.data = 11});
+
+    std::size_t index = 0;
+
+    printf("\n\n### mouse events using .read():");
+    while (true)
+    {
+        static auto mouseReader = EventReader<MyEvent, MyEvent::Mask::MOUSE_EVENT>(pipe, index);
```

Comment:
Can this be moved outside the loop so it's not redefined every time?

Reply:
I think the point was that you could destruct and it and construct it again and it wouldn't make any difference.
(The static shouldn' t be there then).